src/Spectre.libGaussianMixtureModelling.Tests/ExpectationMaximizationTest.cpp

```diff
@@ -0,0 +1,287 @@
+/*
+* ExpectationMaximizationTest.cpp
+* Provides implementation of tests for the purpose of testing
+* Gaussian Mixture Modelling Expectation Maximization algorithm.
+*
+Copyright 2017 Michal Gallus
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#define GTEST_LANG_CXX11 1
+#define _CRT_RAND_S  
+
+#include <gtest/gtest.h>
+#include "ExpectationMaximization.h"
+#include "BasicInitializationRef.h"
+#include "ExpectationRunnerRef.h"
+#include "MaximizationRunnerRef.h"
+#include "LogLikelihoodCalculator.h"
+
+
+namespace Spectre::libGaussianMixtureModelling
+{
+    class ExpectationMaximizationTest : public ::testing::Test
+    {
+    protected:
+
+        std::vector<double> mzs;
+        std::vector<double> intensities;
+        std::vector<GaussianComponent> gaussianComponents;
+
+        virtual void SetUp() override
+        {
+            gaussianComponents = {
+                { -5.0, 3.0, 0.25 },
+                { 5.0, 3.0, 0.25 },
+                { -2.0, 9.0, 0.5 }
+            };
+
+            mzs = GenerateRange(-20.0, 20.0, 0.5);
+            intensities = GenerateValues(mzs, gaussianComponents);
+        }
+
+        std::vector<double> GenerateRange(double start, double end, double step)
+        {
+            const int size = int((end - start) / step);
+            std::vector<double> result(size);
+            std::fill(result.begin(), result.end(), start);
+
+            for (int i = 0; i < size; i++)
+            {
+                result[i] += step * i;
+            }
+
+            return result;
+        }
+
+        std::vector<double> GenerateValues(std::vector<double> input, 
+            const std::vector<GaussianComponent>& components)
```

Comment:
We can only add comments. While C syntax does, C++ syntax doesn't allow for the following:
```
gaussianComponents = {
                { .mean =-5.0, .deviation = 3.0, .weight = 0.25 },
                { .mean = 5.0, .deviation = 3.0, .weight = 0.25 },
                { .mean =-2.0, .deviation = 9.0, .weight = 0.5 }
            };
```